doc/source/guide/in_depth/functional_guide.rst

```diff
@@ -0,0 +1,257 @@
+======================
+It is common in applications to perform arithmetic operations with functionals, for example adding two functionals :math:`f` and :math:`g`
+
+.. math::
+   [f+g](x) = f(x) + g(x),
+
+or multiplication of a functional by a scalar
+
+.. math::
+   [\alpha f](x) = \alpha f (x).
+
+Another example is translating a functional with a vecotr :math:`y`
+
+.. math::
+   f(x - y),
+
+or given an `Operator` :math:`A` whose range is the same as the domain of the functional we also have composition
+
+.. math::
+    [f * A](x) = f(A(x)). 
+
+In some of these cases, properties and methods such as ``gradient``, ``convex_conjugate`` and ``proximal`` can be calculated automatically given a default implementation of the corresponding property in :math:`f`.
+
+All available functional arithmetic, including which properties and methods that automatically can be calculated, is shown below. ``f``, ``g`` represent `Functional`'s, and ``A`` an `Operator` whose range is the same as the domain as the functional. ``a`` is a scalar in the field of the domain of ``f`` and ``g``, and ``y`` is a vector in the domain of ``f`` and ``g``.
+
++--------------------+-----------------+-------------------------------------------------+
+| Code               | Meaning         | Class                                           |
++====================+=================+=================================================+
+| ``(f + g)(x)``     | ``f(x) + g(x)`` | `FunctionalSum`                                 |
+|                    |                 | - Retains `gradient`.                           |
++--------------------+-----------------+-------------------------------------------------+
+| ``(f + a)(x)``     | ``f(x) + a``    | `FunctionalScalarSum`                           |
+|                    |                 | - Retains all properties.                       |
++--------------------+-----------------+-------------------------------------------------+
+| ``(f * A)(x)``     | ``f(A(x))``     | `FunctionalComp`                                |
+|                    |                 | - Retains gradient                              |
++--------------------+-----------------+-------------------------------------------------+
+| ``(a * f)(x)``     | ``a * f(x)``    | `FunctionalLeftScalarMult`                      |
+|                    |                 | - Retains all properties, if ``a`` is positive. |
++--------------------+-----------------+-------------------------------------------------+
+| ``(f * a)(x)``     | ``f(a * x)``    | `FunctionalRightScalarMult`                     |
+|                    |                 | - Retains all properties                        |
++--------------------+-----------------+-------------------------------------------------+
+| ``(v * f)(x)``     | ``v * f(x)``    | `FunctionalLeftVectorMult`                      |
+|                    |                 | - This is not a functional, but an operator     |
```

Comment:
Unclear what "This" means. Perhaps "Results in an operator rather than a functional."